bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusCGateHandler.java

```diff
@@ -0,0 +1,433 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.cbus.CBusBindingConstants;
+import org.openhab.binding.cbus.internal.CBusCGateConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.daveoxley.cbus.CGateConnectException;
+import com.daveoxley.cbus.CGateException;
+import com.daveoxley.cbus.CGateInterface;
+import com.daveoxley.cbus.CGateSession;
+import com.daveoxley.cbus.CGateThreadPool;
+import com.daveoxley.cbus.CGateThreadPoolExecutor;
+import com.daveoxley.cbus.events.EventCallback;
+import com.daveoxley.cbus.status.StatusChangeCallback;
+
+/**
+ * The {@link CBusThreadPool} is responsible for executing jobs from a threadpool
+ *
+ * @author John Harvey - Initial contribution
+ */
+
+@NonNullByDefault
+final class CBusThreadPool extends CGateThreadPool {
+
+    private final Map<@Nullable String, @Nullable CGateThreadPoolExecutor> executorMap = new HashMap<>();
+
+    @Override
+    protected synchronized CGateThreadPoolExecutor CreateExecutor(@Nullable String name) {
+        @Nullable
+        CGateThreadPoolExecutor executor = executorMap.get(name);
+
+        if (executor != null) {
+            return executor;
+        }
+        CGateThreadPoolExecutor newExecutor = new CBusThreadPoolExecutor(name);
+        executorMap.put(name, newExecutor);
+        return newExecutor;
+    }
+
+    @NonNullByDefault
+    public class CBusThreadPoolExecutor extends CGateThreadPoolExecutor {
+        private final ThreadPoolExecutor threadPool;
+
+        public CBusThreadPoolExecutor(@Nullable String poolName) {
+            threadPool = (ThreadPoolExecutor) ThreadPoolManager.getPool("org.openhab.binding.cbus-" + poolName);
+        }
+
+        @Override
+        protected void execute(@Nullable Runnable runnable) {
+            if (runnable != null) {
+                threadPool.execute(runnable);
+            }
+        }
+    }
+}
+
+/**
+ * The {@link CBusCGateHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Scott Linton - Initial contribution
+ */
+
+@NonNullByDefault
+public class CBusCGateHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(CBusCGateHandler.class);
+    private CBusCGateConfiguration configuration = new CBusCGateConfiguration();
+    private @Nullable InetAddress ipAddress = null;
+    public @Nullable CGateSession cGateSession = null;
+    private @Nullable ScheduledFuture<?> keepAliveFuture = null;
+
+    public CBusCGateHandler(Bridge br) {
+        super(br);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Nothing to do here
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.OFFLINE);
+        logger.debug("Initializing CGate Bridge handler.");
+        Optional<CBusCGateConfiguration> cfg = Optional.of(getConfigAs(CBusCGateConfiguration.class));
```

Comment:
There is no reason for you to be using an optional here, the returned config will always be non-null.

Reply:
Ok. I copied that code from another binding but happy to remove it.